src/vuestic-theme/vuestic-components/va-rating/VaStar.vue

```diff
@@ -0,0 +1,133 @@
+<template>
+  <div class="va-star">
+    <i
+      :class="iconClassesComputed"
+      :style="iconStyles"
+      class="va-star__icon"
+      @mousemove="onHover"
+      @mouseleave="removeHover"
+      @click="rate()"
+      width="50"
+    />
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'va-star',
+  props: {
+    value: {
+      type: Boolean
+    },
+    icon: {
+      type: String,
+      default: 'fa fa-star'
+    },
+    isRatingHover: {
+      type: Boolean
+    },
+    halfIcon: {
+      type: String
+    },
+    isHalf: {
+      type: Boolean
+    },
+    iconClasses: {
+      type: String
+    },
+    iconStyles: {
+      type: Object
+    },
+    color: {
+      type: String
+    },
+    emptyIcon: {
+      type: String
+    },
+    hover: {
+      type: Boolean
+    }
+  },
+  data () {
+    return {
+      extraClasses: '',
+      hoverValue: ''
+    }
+  },
+  computed: {
+    iconClassesComputed () {
+      if (this.halfIcon) {
+        if (this.isRatingHover) {
+          if (this.hover) {
+            if (this.hoverValue !== 0.5) {
+              return this.iconClasses + this.icon
+            } else {
+              return this.halfIcon
+            }
+          } else {
+            return this.emptyIcon
+          }
+        } else {
+          if (this.value) {
+            return this.icon
+          } else if (this.isHalf) {
+            return this.halfIcon
+          } else {
+            return this.emptyIcon
+          }
+        }
+      }
+      return this.iconClasses + this.extraClasses
+    }
+  },
+  methods: {
+    rate () {
+      if (this.extraClasses === this.halfIcon) {
+        this.$emit('click', 0.5)
+      } else {
+        this.$emit('click', 1)
+      }
+    },
+    onHover (item) {
+      if (this.halfIcon) {
+        let size = Number(this.iconStyles.fontSize.replace(/[^-0-9]/gim, ''))
```

Comment:
let -> const

> Number(this.iconStyles.fontSize.replace(/[^-0-9]/gim, ''))

Mind to explain what happens here? Comment perhaps?

Reply:
We could set icon sizes in 2 ways:
1) using 3 options: ['large', 'medium', 'small'].
2) using pixes/rems. using this regular expression i put this value to variable size as a number, not as string (without 'px')